src/components/TodoList/TodoList.jsx

```diff
@@ -1,44 +1,93 @@
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+export class TodoList extends React.Component {
+  state = {
+    search: '',
+    showTodo: 'all',
+  }
+
+  filterBySelect = {
+    all: callback => callback,
+    completed: callback => callback.completed,
+    active: callback => !callback.completed,
+  }
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+
+    this.setState({
+      [name]: value,
+    });
+  }
+
+  render() {
+    const { todos, selectedUser, selectedUserId } = this.props;
+    const { search, showTodo } = this.state;
+
+    const filterTodos = todos
+      .filter(todo => todo.title.toLowerCase().includes(search.toLowerCase())
+        && this.filterBySelect[showTodo](todo));
+
+    return (
+      <div className="TodoList">
+        <h2>Todos:</h2>
+        <SearchInput
+          showTodo={showTodo}
+          search={search}
+          handleChange={this.handleChange}
+        />
+
+        <div className="TodoList__list-container">
+          <ul className="TodoList__list">
+            {filterTodos.map(({ id, userId, completed, title }) => (
+              <li
+                className={classNames('TodoList__item',
+                  {
+                    'TodoList__item--unchecked': !completed,
+                    'TodoList__item--checked': completed,
+                  })}
+                key={id}
+              >
+                <label>
+                  <input
+                    type="checkbox"
+                    checked={completed}
+                    readOnly
+                  />
+                  <p>{title}</p>
+                </label>
+                <button
+                  className={classNames('button TodoList__user-button', {
+                    // eslint-disable-next-line max-len
+                    'TodoList__user-button--selected': userId === selectedUserId,
+                  })}
+                  type="button"
+                  onClick={() => selectedUser(userId)}
+                >
+                  {`UserId #${userId}`}
+                </button>
```

Comment:
only one button should be selected

![image](https://user-images.githubusercontent.com/57598825/98437864-89bb7080-20ee-11eb-87ea-b34f8f8b04d4.png)